src/com/timedfly/TimedFly.java

```diff
@@ -25,244 +24,185 @@
-        Bukkit.getServer().getPluginManager().registerEvents(new FlightTime(this.nms, this.flyGUI, this.utility, this.languages), this);
-        Bukkit.getServer().getPluginManager().registerEvents(new Inventory(currencyManager, this.utility, this.languages, this.itemsConfig, this.nms), this);
-        Bukkit.getServer().getPluginManager().registerEvents(new CustomFlyCMD(this.utility, this.languages, this.flyGUI, this.nms), this);
-        Bukkit.getServer().getPluginManager().registerEvents(new JoinLeave(this.utility, sqlManager, this, this.updater), this);
-        Bukkit.getServer().getPluginManager().registerEvents(new ChangeWorld(this.utility, this), this);
-        Bukkit.getServer().getPluginManager().registerEvents(new ChangeGameMode(this.utility), this);
-        Bukkit.getServer().getPluginManager().registerEvents(new Respawn(this.utility, this), this);
-        Bukkit.getServer().getPluginManager().registerEvents(new PlayerOnGround(this.utility), this);
-        if (version.startsWith("v1_8")) {
-            Bukkit.getServer().getPluginManager().registerEvents(new FallDamage(), this);
-        }
-
-        if (version.equals("v1_12_R1")) {
-            Bukkit.getServer().getPluginManager().registerEvents(new AttackMob(this, this.utility, this.languages), this);
-        }
-
+        CurrencyManager currencyManager = new CurrencyManager(tokensManager, playerPointsHook, vault, languages);
+        Bukkit.getServer().getPluginManager().registerEvents(new FlightTime(nms, flyGUI, utility, languages), this);
+        Bukkit.getServer().getPluginManager().registerEvents(new Inventory(currencyManager, utility, languages, itemsConfig, nms), this);
+        Bukkit.getServer().getPluginManager().registerEvents(new CustomFlyCMD(utility, languages, flyGUI, nms), this);
+        Bukkit.getServer().getPluginManager().registerEvents(new JoinLeave(utility, sqlManager, this, updater), this);
+        Bukkit.getServer().getPluginManager().registerEvents(new ChangeWorld(utility, this), this);
+        Bukkit.getServer().getPluginManager().registerEvents(new ChangeGameMode(utility), this);
+        Bukkit.getServer().getPluginManager().registerEvents(new Respawn(utility, this), this);
+        Bukkit.getServer().getPluginManager().registerEvents(new PlayerOnGround(utility), this);
+
+        if (version.startsWith("v1_8")) Bukkit.getServer().getPluginManager().registerEvents(new FallDamage(), this);
+        if (version.equals("v1_12_R1"))
+            Bukkit.getServer().getPluginManager().registerEvents(new AttackMob(this, utility, languages), this);
     }
 
     private boolean setupNMS() {
         try {
-            version = this.getServer().getClass().getPackage().getName().replace(".", ",").split(",")[3];
-        } catch (ArrayIndexOutOfBoundsException var3) {
+            version = getServer().getClass().getPackage().getName().replace(".", ",").split(",")[3];
+        } catch (ArrayIndexOutOfBoundsException whatVersionAreYouUsingException) {
```

Comment:
ArrayIndexOutOfBoundsException => whatVersionAreYouUsingException ???
bad wording

Reply:
True, but I don't think it matters that much right now since its not being used.